### PR TITLE
Update API reference doc for Pipes

### LINKED ISSUE #33897: date pipe documentation is misleading, the date pipe is a pure pipe
Update: See https://github.com/angular/angular/issues/33897#issuecomment-555440953 for info about the docs issue.

Original issue:
-----

# 🐞 bug report

### Affected Package

@angular/core

### Is this a regression?

 Yes, the previous versions work fine 

### Description
I have two properties in TS which I render in html for the first time with ok, then with function(TS) I change values but in html there is no change.  

## 🔬 Minimal Reproduction

 https://stackblitz.com/edit/angular-7gifwf

## 🔥 Exception or Error
<pre><code>

</code></pre>

## 🌍  Your Environment

**Angular Version:**
<pre><code>

</code></pre>

**Anything else relevant?**

### PR DESCRIPTION
**Pull request by jbogarthyde**
Edit descriptions, usage examples, and add links to be complete and consistent with API reference doc style

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?
Pipe API doc lacks links back to guide material (Pipes and Internationalization guides)

Issue Number: N/A

## What is the new behavior?
Language/style edits, add links to guides.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by petebacondarwin — CHANGES_REQUESTED** [review]
Reviewed-for: fw-i18n

**Review by petebacondarwin — APPROVED** [review]

**Comment by jbogarthyde** [comment]
Pete has approved this. Can someone from FW please do pull-approve?

**Review by atscott — APPROVED** [review]
reviewed-for: fw-core

**Comment by jbogarthyde** [comment]
@gkalpak Can you do global approve and G3? (@atscott has approved, although git is still listing him as pending).

**Review by atscott — APPROVED** [review]

**Comment by jbogarthyde** [comment]
This is docs only, should not need G3

**Comment by AndrewKushnir** [comment]
Hi @jbogarthyde, it looks like @gkalpak or @IgorMinar should provide the final approval before this PR gets merged, so I'm removing the "merge" label for now. Feel free to add it back once Igor or George approve the changes. Thank you.

File: aio/content/guide/deprecations.md [hunk 1]
```diff
@@ -75,14 +74,6 @@ Tip: In the [API reference section](api) of this doc site, deprecated APIs are i
...
 </div>
 
-{@a common}
-### @angular/common
-
-| API                                                                                           | Replacement                                         | Deprecation announced | Notes |
-| --------------------------------------------------------------------------------------------- | --------------------------------------------------- | --------------------- | ----- |
-| [`CurrencyPipe` - `DEFAULT_CURRENCY_CODE`](api/common/CurrencyPipe#currency-code-deprecation) | `{provide: DEFAULT_CURRENCY_CODE, useValue: 'USD'}` | v9                    | From v11 the default code will be extracted from the locale data given by `LOCAL_ID`, rather than `USD`. |
-
-
```
Comment (IgorMinar) [inline aio/content/guide/deprecations.md:85]: hey @jbogarthyde can you please help me understand why this section is being removed?

The deprecation is still active, so as far as I can tell, this text should remain here. Am I missing something? thanks

**Review by IgorMinar — DISMISSED** [review]
Reviewed-for: docs-packaging-and-releasing

File: aio/content/guide/deprecations.md (hunk 1 above)
Comment (jbogarthyde) [reply aio/content/guide/deprecations.md:85]: I don't remember why I removed it, so I'm restoring it.

**Comment by jbogarthyde** [comment]
@IgorMinar Deprecation info restored, please recheck.

**Review by jelbourn — APPROVED** [review]
LGTM, using global approvers because this file touches the deprecations doc, but it only removes trailing whitespace. 

Reviewed-for: global-approvers

**Pull request closed without merging**